### PR TITLE
encounter.diagnosis fix

### PR DESCRIPTION
**Pull request by xpander53**

### CODE REVIEW

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java [hunk 1]
```diff
@@ -258,9 +260,85 @@ public void validateEncounterMultipleDiagnoses() {
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java:260]: Please add a check like this:

        // Verify reasonReference has a reference for each condition.
        List<Base> reasonReferences = encounter.getNamedProperty("reasonReference").getValues();
        // Therefore there should be 1.
        assertThat(reasonReferences).hasSize(1);

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java [hunk 2]
```diff
@@ -248,7 +249,8 @@ public void validateEncounterMultipleDiagnoses() {
 
         Coding condCoding = condCC.getCoding().get(0);
         assertThat(condCoding.hasSystem()).isTrue();
-        // change from http://hl7.org/fhir/sid/icd-10 to http://hl7.org/fhir/sid/icd-10-cm temporarily, see Issue #189
+        // change from http://hl7.org/fhir/sid/icd-10 to
+        // http://hl7.org/fhir/sid/icd-10-cm temporarily, see Issue #189
         assertThat(condCoding.getSystem()).isEqualTo("http://hl7.org/fhir/sid/icd-10-cm");
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java:254]: Can we also a check that condition.encounter is set correctly?

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java (hunk 1 above)
Comment (xpander53) [reply src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java:260]: Added

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java (hunk 2 above)
Comment (xpander53) [reply src/test/java/io/github/linuxforhealth/hl7/segments/HL7ConditionFHIRConversionTest.java:254]: added

**Pull request merged**